src/components/core/compute/initialize.ts

```
@@ -81,6 +82,16 @@ export class ComputeInitializeHandler extends Handler {
               }
             }
           }
+          if (ddo && ddo.nft.state && isOrderingAllowedForAsset(ddo)) {
```

Hi Maria
It seems you have the logic inverted on all the IF blocks :-)
we have:
```
if (ddo && ddo.nft.state && isOrderingAllowedForAsset(ddo)) {
...
// return some error
} 
```
We actually want to return an error IF  =>  ` ! isOrderingAllowedForAsset(ddo) ` (if this fn returns false)

One other thing, is that we're also not considering that `ddo.nft` might be `undefined` , so every time we have like  `ddo.nft.state`  it can potentially break..

IMO, it would be a lot more easier if all the IFs were simply re-written  like:
```
if (! isOrderingAllowedForAsset(ddo)) {
...
// return some error
}
```

Then we can check all the necessary conditions inside` isOrdedingAllowedForAsset() `function, without repeating same lines everywhere... like for example:
```
export function isOrderingAllowedForAsset(asset: DDO): boolean {
  if (!asset) {
    return false
  } else if (
    asset.nft &&
    !(asset.nft.state in [MetadataStates.ACTIVE, MetadataStates.UNLISTED])
  ) {
    return false
  }

  return true
}
```
thanks

---

sure, I'll modify accordingly.

---

I'll also add to the isOrderingAllowedForAsset, the reason for possible errors in order to be logged.

```javascript

export function isOrderingAllowedForAsset(asset: DDO): OrdableAssetResponse {
  if (!asset) {
    return { 
       isOrdable: false,
       reason: `Asset provided is either null, either undefined ${asset}`
    }
  } else if (
    asset.nft &&
    !(asset.nft.state in [MetadataStates.ACTIVE, MetadataStates.UNLISTED])
  ) {
    return { 
       isOrdable: false,
       reason: "Nft not present in the asset or the state is different than ACTIVE or UNLISTED."
    }
  }

  return {
    isOrdable: true,
    reason: ''
  }
}
```